daemon-scheduler/pkg/engine/monitor.go

```diff
@@ -48,13 +50,18 @@ func NewMonitor(
 	}
 }
 
-func (m monitor) InProgressMonitorLoop() {
-	ticker := time.NewTicker(inProgressMonitorTickerDuration)
+func (m monitor) InProgressMonitorLoop(tickerDuration time.Duration) {
```

Comment:
because if the ticker duration is hard coded it cannot be modified in unit tests and they end up taking 10 secs per test, which is unnecessary and makes build very long.